tests/src/python/test_qgsserver_wfs.py

```diff
@@ -189,7 +189,7 @@ def test_wfs_getcapabilities_100_url(self):
                 self.assertEqual(
                     "onlineResource=\"my_wfs_advertised_url\"" in item, True)
 
-    def result_compare(self, file_name, error_msg_header, header, body):
+    def result_compare(self, file_name, error_msg_header, header, body, expectEqual=True):
```

Comment:
I don't really understand the `expectEqual` here, what is it for?

Reply:
I want to test if two requests with the same coordinate but different EPSG return different (unit test 1 and 2 from the description). But we don't have an `assertXMLNotEqual`, so I added `expectEqual` to the result compare. By default, it will pass if the XML is equal. If it is set to False, the XML shouldn't be the same.

It's only used for this one `withinFake3857FilterTemplate_post`

Reply:
Cannot use assertRaise for testing an expected failure?

Reply:
Ah, yes, I think it's a cleaner way to do it. Let me update it.